### DISCUSSION TITLE
Remove duplicates based on column number

### DISCUSSION THREAD

**Discussion by johncwelch** [dfinke/ImportExcel#1552] [Q&A]
Is there a way with ImportExcel to deduplicate based on column values? 

If you're scripting excel directly, then it would be say $theWorkSheet.UsedRange.RemoveDuplicates(columnumber)

Thanks!

**Comment by semibreave** [comment]
let's say 

<html>
<head>
	
	
	<title></title>
	<meta name="generator" content="LibreOffice 7.6.0.3 (Windows)"/>
	<style type="text/css">
		body,div,table,thead,tbody,tfoot,tr,th,td,p { font-family:"Liberation Sans"; font-size:x-small }
		a.comment-indicator:hover + comment { background:#ffd; position:absolute; display:block; border:1px solid black; padding:0.5em;  } 
		a.comment-indicator { background:red; display:inline-block; border:1px solid black; width:0.5em; height:0.5em;  } 
		comment { display:none;  } 
	</style>
	
</head>

<body>

A | B
-- | --
alice | 20
john | 33
alice | 20
ahmad | 43
john | 33


</body>

</html>

$obj = Import-Excel .\discuss.xlsx
$obj | Group-Object A,B | ForEach-Object { $_.Group[0] }

Output:


A       B
-       -
alice  20
john   33
ahmad  43

**Reply by johncwelch** [reply]
thanks! I'll give that a try!

**Comment by dfinke** [comment]
@johncwelch Thanks for the question.

Does this exist? Or it is how you'd like to be able to use it?

```powershell
$theWorkSheet.UsedRange.RemoveDuplicates(columnumber)
```

**Reply by johncwelch** [reply]
a little column a, a little column b? It looks like based on semibreave's answer that it's possible, but the syntax I used as a sample would be much easier to work with, and be a good fit to Open-ExcelPackage.
